proposals/secure-storage.md

```diff
@@ -0,0 +1,158 @@
+
+**What’s the plan for Linux?**
+
+Linux is one of the platforms where this API would be most beneficial, so we’d love to support it. However, it’s less clear which API would be appropriate here. APIs such as the [FreeDesktop Secure Storage API](https://www.gnu.org/software/emacs/manual/html_node/auth/Secret-Service-API.html) exist, but not all of the available APIs are guaranteed to be hardware backed and other options have different limitations. Feedback is welcome on what should be done here.
+
+**Why is it useful to allow storing secrets without biometric authentication?**
+
+Millions of users use devices without the hardware for biometric authentication, but install extensions which would be likely to implement this API. Ensuring the API is still available would allow us to implement things like PIN code unlock more securely. This wouldn’t be quite as secure as a biometric approach, but on other platforms we’ve found that this fits within our threat model and drastically improves user experience with very little effort.
+
+**Are there any alternatives? How does 1Password implement biometrics today?**
+
+We currently use a solution based on Native Messaging where the extension communicates with a native app. This app is able to use system APIs to request biometrics and access hardware based storage. This has limitations, however. Not all users have the native application (users of Chromebooks and iOS Safari, to name a few) and implementing reliable cross-process messaging is a challenge.
+
+**Could secrets be stored in localStorage, IDB or the new storage.session instead?**
+
+We’ve historically chosen not to do this because it makes the secrets too easy to access. While a compromised machine is hard to protect against, data in these APIs can be trivially accessed with a console command and no authentication. This allows an “evil maid” attack if a computer is left unlocked, or for another application to read the secret from disk independently of Chrome. While the secret must be in memory while the extension is in use, extensions like password managers aggressively enter a “locked state” after inactivity where secrets are cleared from memory. This means that after only a short period of idle time such an attack would require biometric authentication.
+
+## Remaining Work
+
+- We need to gather feedback from the community and browser vendors to decide which proposal to move forward with, and if any changes are needed.
+- It would be beneficial to support more fine-grained control of the keys generated in Proposal 1. We should document this in more detail, perhaps drawing inspiration from the [Web Crypto API](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/generateKey).
+
+## Contacts
+
+Oliver Dunk (oliver@1password.com)
```

Comment:
Let's prepend a section "Champions" before the Motivation section, wher the list of people driving the proposal is shown. And include the Github handle, to make it easier for readers/contributors to tag the right person on Github.

Reply:
I like that! I went ahead and tweaked things as suggested.